tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/method/TeamMailboxMemberSetMethod.scala

```diff
@@ -0,0 +1,138 @@
+      .map(response => Invocation(
+        methodName = methodName,
+        arguments = Arguments(Serializer.serializeSetResponse(response).as[JsObject]),
+        methodCallId = invocation.invocation.methodCallId))
+      .map(InvocationWithContext(_, invocation.processingContext))
+
+  private def performUpdates(username: Username, request: TeamMailboxMemberSetRequest): SMono[TeamMailboxMemberSetResponse] =
+    SFlux.fromIterable(request.validatedUpdateRequest())
+      .flatMap(parsingRequestResult => performUpdate(username, parsingRequestResult), ReactorUtils.LOW_CONCURRENCY)
+      .collectSeq()
+      .map(listResult => TeamMailboxMemberSetResponse.from(request.accountId, listResult))
+
+  private def performUpdate(username: Username, parsingRequestResult: TeamMailboxMemberParsingRequestResult): SMono[TeamMailboxMemberSetResult] =
+  {
+    val r: SMono[TeamMailboxMemberSetResult] = for {
+      requestWithValidSyntax <- validateSyntax(parsingRequestResult)
+      requestWithUserExistenceValidation <- validateUsersExist(requestWithValidSyntax)
+      presentMembers <- retrievePresentMembers(requestWithUserExistenceValidation)
+      requestWithAuthorizationValidation <- validateAuthorizations(username, requestWithUserExistenceValidation, presentMembers)
+      requestWithAuthorizationValidation <- validateNoChangeToManagers(requestWithAuthorizationValidation, presentMembers)
+      result <- updateMembers(requestWithAuthorizationValidation)
+    } yield {
+      result
+    }
+    r.onErrorResume {
+      case e: NonExistingUsers => SMono.just(TeamMailboxMemberSetResult.notUpdated(e.teamMailbox,
+        SetErrorDescription(s"Some users do not exist in the system: ${e.usersNotFound.map(username => username.asString()).toArray.mkString("", ", ", "")}")))
+      case e: NotAuthorized => SMono.just(TeamMailboxMemberSetResult.notUpdated(e.teamMailbox, SetErrorDescription(s"Not manager of teamMailbox ${e.teamMailbox.asString()}")))
+      case e: TeamMailboxNotFoundException => SMono.just(TeamMailboxMemberSetResult.notUpdated(e.teamMailbox, SetErrorDescription("Team mailbox is not found or not a member of the mailbox")))
+      case e: ManagerAuthorizationChanged => SMono.just(TeamMailboxMemberSetResult.notUpdated(e.teamMailbox, SetErrorDescription(s"Could not update or remove manager ${e.username.asString()}")))
+      case e: InvalidSyntax => SMono.just(TeamMailboxMemberSetResult.notUpdated(e.failure.teamMailboxName, SetErrorDescription(e.failure.exception.message)))
+    }
+  }
+
+  private def retrievePresentMembers(requestWithUserExistenceValidation: ParsingRequestSuccess): SMono[Map[Username, TeamMailboxMember]] =
+    SFlux(teamMailboxRepository.listMembers(requestWithUserExistenceValidation.teamMailbox)).collectMap[Username, TeamMailboxMember](member => member.username, member => member)
+
+  private case class NonExistingUsers(teamMailbox: TeamMailbox, usersNotFound: Seq[Username]) extends RuntimeException
```

Comment:
Benoit's code is just a draft. It's not a big convention deal, but we can create an object TeamMailboxMemberSetMethod and move the exception case class into it.